app/models/jobs_json_request_handler.rb

```diff
@@ -0,0 +1,65 @@
+class JobsJsonRequestHandler
+  attr_reader :controller, :params, :response, :filter_id, :cluster_id
+
+  def initialize(filter_id:, cluster_id:, controller:, params:, response:)
+    @filter_id = filter_id
+    @cluster_id = cluster_id
+    @controller = controller
+    @params = params
+    @response = response
+  end
+
+  def filter
+    @filter ||= Filter.list.find(Filter.all_filter) { |f| f.filter_id == filter_id  }
+  end
+
+  def clusters
+    if cluster_id == 'all'
+      OODClusters
+    else
+      OODClusters.select { |c| c == cluster_id}
+    end
+  end
+
+  def render
+    controller.render :json => get_jobs
+  end
+
+  # Get a set of jobs defined by the filtering parameter.
+  def get_jobs
+    jobs = Array.new
+    errors = Array.new
+
+    clusters.each do |cluster|
+      begin
+        if filter.user?
+          result = cluster.job_adapter.info_where_owner(OodSupport::User.new.name)
+        else
+          result = filter.apply(cluster.job_adapter.info_all)
+        end
+
+        # Only add the running jobs to the list and assign the host to the object.
+        #
+        # There is also curently a bug in the system where jobs with an empty array
```

Comment:
This sounds like it is specific to Torque?

Reply:
...or possibly LSF and PBSPro. As we discussed, I realized that I haven't looked at either of those systems in detail, and it has been a while since I have used either.